src/lib.rs

```diff
@@ -522,6 +541,30 @@ mod tests {
         }
     }
 
+    test_beat! {
+        name: arr_pushes_zero,
+        text: "1 2 3 [0",
+        code: [Num(1.0), Num(2.0), Num(3.0), Arr(0)],
+        eval: {
+            0.0 => 0.0,
+            1.0 => 0.0,
+            2.0 => 0.0,
+            3.0 => 0.0,
+            4.0 => 0.0,
+        }
+    }
+
+    test_beat! {
+        name: empty_arr,
```

Comment:
Same for this, this is a static failure not a dynamic failure and so should produce an error on evaluation.